### PR TITLE
[fix](multi-catalog)fix maxcompute partition filter and session creation

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

add maxcompute partition support
fix maxcompute partition filter
modify maxcompute session create method

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java [hunk 1]
```diff
@@ -124,6 +129,7 @@ protected void initTableInfo(ColumnType[] requiredTypes, String[] requiredFields
         }
         // reorder columns
         List<Column> columnList = curTableScan.getSchema().getColumns();
+        columnList.addAll(curTableScan.getSchema().getPartitionColumns());
```
Comment (morningman) [inline fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java:132]: Does `curTableScan.getSchema().getColumns()` contains partition columns?

File: fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java [hunk 2]
```diff
@@ -139,13 +145,24 @@ public void open() throws IOException {
...
             long recordCount = session.getRecordCount();
             totalRows = splitSize > 0 ? Math.min(splitSize, recordCount) : recordCount;
 
             arrowAllocator = new RootAllocator(Long.MAX_VALUE);
-            curReader = session.openArrowRecordReader(start, totalRows, readColumns, arrowAllocator);
+            Set<String> partitionColumns = session.getSchema().getPartitionColumns().stream()
+                    .map(Column::getName)
+                    .collect(Collectors.toSet());
+            List<Column> maxComputeColumns = new ArrayList<>(readColumns);
+            maxComputeColumns.removeIf(e -> partitionColumns.contains(e.getName()));
```
Comment (morningman) [inline fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java:164]: Why remove partition columns here?

File: regression-test/suites/external_table_p2/maxcompute/test_external_catalog_maxcompute.groovy [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            qt_q2 """ select count(*) from store_sales """
+            qt_q3 """ select count(*) from mc_parts """ // partition table test
+        }
+        // type test
+        def q02 = {
+            qt_q4 """ select count(*) from int_types """ // test bool,tinyint,int,bigint
+        }
+        // test partition table filter
+        def q03 = {
+            qt_q5 """ select mc_string, dt from mc_parts where dt = 'dts' """
```
Comment (morningman) [inline regression-test/suites/external_table_p2/maxcompute/test_external_catalog_maxcompute.groovy:48]: We need to add more complex cases, including:
1. all column types.
2. partition tables
3. complex where predicate including and/or/in combination

File: regression-test/suites/external_table_p2/maxcompute/test_external_catalog_maxcompute.groovy [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_external_catalog_maxcompute", "p2,external,maxcompute,external_remote,external_remote_maxcompute") {
+    String enabled = context.config.otherConfigs.get("enableMaxComputeTest")
+    if (enabled != null && enabled.equalsIgnoreCase("true")) {
+        String ak = getS3AK()
```
Comment (morningman) [inline regression-test/suites/external_table_p2/maxcompute/test_external_catalog_maxcompute.groovy:21]: Where to get ak and sk? It should be aliyun's ak and sk?

File: fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java (hunk 2 above)
Comment (wsjz) [reply fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java:168]: max compute depart the partition columns from common columns

File: fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java (hunk 1 above)
Comment (wsjz) [reply fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java:137]: not contains

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run external

**Comment by wsjz** [comment]
run external

**Comment by wsjz** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by zhangguoqiang666 — APPROVED** [review]
LGTM

**Pull request merged**